source/jormungandr/jormungandr/scenarios/tests/journey_compare_tests.py

```diff
@@ -1151,3 +1151,50 @@ def test_activate_deactivate_min_taxi():
 
     f = jf.FilterTooShortHeavyJourneys(min_taxi=8, orig_modes=['taxi'], dest_modes=['taxi', 'walking'])
     assert not f.filter_func(journey)
+
+
+def test_filter_direct_path_mode_car():
+    # self.dp == 'none' and is_dp
+    journey = response_pb2.Journey()
+    journey.tags.append("car")
+    journey.tags.append("non_pt")
+    f = jf.FilterDirectPathMode(dp="none", dp_mode=[], orig_modes=[], dest_modes=[])
+    assert not f.filter_func(journey)
+
+    # self.dp == 'only' and not is_dp
+    journey = response_pb2.Journey()
+    journey.tags.append("car")
+    f = jf.FilterDirectPathMode(dp="only", dp_mode=[], orig_modes=[], dest_modes=[])
+    assert not f.filter_func(journey)
+
+    # is_dp and is_in_direct_path_mode_list
+    journey = response_pb2.Journey()
+    journey.tags.append("car")
+    journey.tags.append("non_pt")
+    f = jf.FilterDirectPathMode(dp="indifferent", dp_mode=["car"], orig_modes=[], dest_modes=[])
+    assert f.filter_func(journey)
+
+    # is_dp and not is_in_direct_path_mode_list
+    journey = response_pb2.Journey()
+    journey.tags.append("car")
+    journey.tags.append("non_pt")
+    f = jf.FilterDirectPathMode(dp="indifferent", dp_mode=["bike"], orig_modes=["car"], dest_modes=[])
+    assert not f.filter_func(journey)
```

Comment:
To me it's not the responsibility of the filter you created.
The journey should just never be produced, not being filtered by this filter.